s3prl/upstream/byol_a/expert.py

```diff
@@ -29,43 +29,36 @@ def __init__(
         self,
         ckpt: str,
         model_config: str,
-        feature_d: int,
-        window_secs: float = 1.0,
-        stride_secs: float = 1.0,
+        norm_mean: float = None,  # Has to be a float value to continue training.
+        norm_std: float = None,  # The same as above.
     ):
         super().__init__()
         config = load_yaml_config(model_config)
 
-        self.window_secs = window_secs
-        self.stride_secs = stride_secs
-        self.output_dim = feature_d
-        self.seg_input_length = len(
-            range(0, int(window_secs * SAMPLE_RATE), config.hop_length)
-        )
-
         # Preprocessor and normalizer.
-        self.to_melspec = torchaudio.transforms.MelSpectrogram(
-            sample_rate=config.sample_rate,
-            n_fft=config.n_fft,
-            win_length=config.win_length,
-            hop_length=config.hop_length,
-            n_mels=config.n_mels,
-            f_min=config.f_min,
-            f_max=config.f_max,
-        )
-        stats = [
-            -5.4919195,
-            5.0389895,
-        ]  # FIXME: should use downstream dataset statistics
-        self.normalizer = PrecomputedNorm(stats)
+        self.to_logmelspec = LogMelSpectrogram()
+        if norm_mean is None or norm_std is None:
+            print('  ** CAUTION **')
+            print('  This is a run for calculating statistics of the downstream task using RunningNorm and will exit in the middle of training. **')
+            print('  ** CAUTION **')
+            self.normalizer = RunningNorm(epoch_samples=10_000, max_update_epochs=1, axis=[0, 1, 2]) # Use single scalar mean/std values.
+        else:
+            print(f'*** Using normalization statistics: mean={norm_mean}, std={norm_std} ***')
+            self.normalizer = lambda x: (x - norm_mean) / norm_std
```

Comment:
Let's use the logger package instead of printing.

```python3
import logging
logger = logging.getLogger(__name__)

logger.info("some information")
```

Reply:
OK, I will follow it.